request.put_test.go

```diff
@@ -0,0 +1,50 @@
+package braid
+
+import (
+	"context"
+	"testing"
+
+	"github.com/stretchr/testify/require"
+)
+
+func TestPutRequestRoundTrip(t *testing.T) {
+	patch1 := []byte(`{"asdf": "jkl;"}`)
+	patch2 := []byte(`{"braid": "http", "oof": ["rab", "zab"]}`)
+
+	expected := &PutRequest{
+		ContentType: "application/json",
+		Accept:      "application/json",
+		Version:     "12345",
+		Parents:     []string{"foo", "bar"},
+		Patches: []Patch{
+			{
+				Name:          "patch-type-1",
+				ContentRange:  "json [-0:-0]",
+				ContentLength: uint64(len(patch1)),
+				ExtraHeaders: map[string]string{
+					"Quux":  "xyzzy",
+					"Quack": "duck",
+				},
+				Body: patch1,
+			},
+			{
+				Name:          "patch-type-2",
+				ContentRange:  "json .foo.bar",
+				ContentLength: uint64(len(patch2)),
+				ExtraHeaders: map[string]string{
+					"Encoding": "flarf",
+					"Cache":    "zork",
+				},
+				Body: patch2,
```

Comment:
This looks so much like how I'm doing things in JS-land!

One difference is I'm calling this one field `Content`, and reserving `Body` for the case where we transmit a whole snapshot without any patches. In my mind I thought it made sense because we have `ContentRange` and `ContentLength` above, which made me think that the actual data is the `Content`.

Hmm.. another way to disambiguate this is as `PatchBody`.